ls1intum/Artemis
# `Plagiarism checks`: Improve email subject for new plagiarism notification

### Checklist
#### General


#### Server

### Motivation and Context



This PR is a follow-up of [#6174](https://github.com/ls1intum/Artemis/pull/6174). The email subject of the new plagiarism case does not provide any information regarding the exercise and course in which the plagiarism case happened. To make it clear to the student, the new plagiarism case email subject should provide more information about the case.

### Description


Add course name and exercise name to the new plagiarism case email subject

### Steps for Testing

**For a detailed example of testing, please see this screencast:**
https://user-images.githubusercontent.com/10885503/217796578-1e0c1940-e80c-4b5d-9c7a-e17c43ec5eee.mov


Prerequisites:
- Test Server 0
- 1 Programming Exercise with the same code submissions to trigger the plagiarism (e.g. [Test 6215](https://artemis-test0.artemis.in.tum.de/courses/5/exercises/80))
- 1 Student with your email

1. Log in as the user with your email
2. Navigate to Exercise [Test 6215](https://artemis-test0.artemis.in.tum.de/courses/5/exercises/80)
3. Click Start Exercise
4. Click Open code editor, then click Open-graded code editor
5. Click BubbleSort.java
6. Insert bubble sort code inside the performSort method: 
```
for (int i = input.size() - 1; i >= 0; i--) {
    for (int j = 0; j < i; j++) {
        if (input.get(j).compareTo(input.get(j + 1)) > 0) {
            Date temp = input.get(j);
            input.set(j, input.get(j + 1));
            input.set(j + 1, temp);
        }
    }
}
```
7. Click Submit
8. Log in to Artemis as an Instructor
9. Navigate to Exercise [Test 6215](https://artemis-test0.artemis.in.tum.de/course-management/5/programming-exercises/80/plagiarism)
10. Click **Rerun plagiarism detection**
11. Click the result, then click Confirm
12. Click the course name of the breadcrumb navigation
13. Click Plagiarism Cases
14. Scroll to Exercise Test 6215
15. Click the case of the account with your email
16. Click Notify Student
17. Click Save
18. Check your email and verify that the email subject contains the course & exercise name

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Branch | Line |
|------------|-------:|-----:|
| MailService.java | 93% | 98% |

### Screenshots


**Before**
<img width="1183" alt="image" src="https://user-images.githubusercontent.com/10885503/217498160-1250522a-cf32-4299-8870-0e9a385e3705.png">

**After**
<img width="1181" alt="image" src="https://user-images.githubusercontent.com/10885503/217796368-e461d51a-13c9-44fe-9cc3-b40396c8283e.png">
